packages/beacon-node/src/node/utils/interop/state.ts

```diff
@@ -34,7 +33,7 @@ export function getInteropState(
   const fork = config.getForkName(GENESIS_SLOT);
   const executionPayloadHeaderType =
     fork !== ForkName.phase0 && fork !== ForkName.altair
-      ? ssz.allForksExecution[fork as ExecutionFork].ExecutionPayloadHeader
+      ? sszTypesFor<ForkExecution>(fork).ExecutionPayloadHeader
```

Comment:
can't we use the type guard `isForkExecution` one line above to properly infer the fork here?

Reply:
actually this is not even needed, already inferred correctly, I would suggest we remove the explicit type

```suggestion
      ? sszTypesFor(fork).ExecutionPayloadHeader
```